lib/Qandidate/Toggle/Serializer.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the qandidate/toggle package.
+ *
+ * (c) Qandidate.com <opensource@qandidate.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Qandidate\Toggle;
+
+/**
+ * Representation of a serializer
+ *
+ * This class provides common ground for concrete serializer implementations
+ *
+ * @TODO: serialize() and deserialize() should be part of abstract interface
+ */
+abstract class Serializer
```

Comment:
This class is currently only introduces to re-use `assertHasKey()`. I'd rather move the `assertHasKey()` method to a utility function / static function in a helper class somewhere. The `@todo` mentions making `serialize()` and `deserialize()` part of the abstract interface, but the signature will be different per implementation, so that doesn't make sense?

Unless there are other arguments for introducing this base class I'd prefer to not introduce it.